pkg/ext/stores/useractivity/store.go

```diff
@@ -0,0 +1,258 @@
+package useractivity
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	ext "github.com/rancher/rancher/pkg/apis/ext.cattle.io/v1"
+	v3Legacy "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	v3 "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/rancher/pkg/wrangler"
+	extcore "github.com/rancher/steve/pkg/ext"
+	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+const (
+	UserActivityNamespace = "cattle-useractivity-data"
+	tokenUserId           = "authn.management.cattle.io/token-userId"
+	SingularName          = "useractivity"
+	PluralName            = "useractivities"
+)
+
+// +k8s:openapi-gen=false
+// +k8s:deepcopy-gen=false
+type Store struct {
+	tokenController v3.TokenController
+	checker         userHandler
+}
+
+var GV = schema.GroupVersion{
+	Group:   "ext.cattle.io",
+	Version: "v1",
+}
+
+var GVK = schema.GroupVersionKind{
+	Group:   GV.Group,
+	Version: GV.Version,
+	Kind:    "UserActivity",
+}
+var GVR = schema.GroupVersionResource{
+	Group:    GV.Group,
+	Version:  GV.Version,
+	Resource: PluralName,
+}
+
+func NewFromWrangler(wranglerCtx *wrangler.Context) *Store {
+	return &Store{
+		tokenController: wranglerCtx.Mgmt.Token(),
+		checker:         &tokenChecker{},
+	}
+}
+
+// GroupVersionKind implements [rest.GroupVersionKindProvider]
+func (t *Store) GroupVersionKind(_ schema.GroupVersion) schema.GroupVersionKind {
+	return GVK
+}
+
+// NamespaceScoped implements [rest.Scoper]
+func (t *Store) NamespaceScoped() bool {
+	return false
+}
+
+// GetSingularName implements [rest.SingularNameProvider]
+func (t *Store) GetSingularName() string {
+	return SingularName
+}
+
+// New implements [rest.Storage]
+func (t *Store) New() runtime.Object {
+	obj := &ext.UserActivity{}
+	obj.GetObjectKind().SetGroupVersionKind(GVK)
+	return obj
+}
+
+// Destroy implements [rest.Storage]
+func (t *Store) Destroy() {
+}
+
+// Create implements [rest.Creator]
+func (uas *Store) Create(ctx context.Context,
+	obj runtime.Object,
+	createValidation rest.ValidateObjectFunc,
```

Comment:
check `createValidation` for Create/Update. See [here](https://github.com/rancher/rancher/blob/ak-imperative-token-resource/pkg/ext/stores/tokens/tokens.go#L268) how it is implemented for tokens

Reply:
@raulcabello should I define this function somewhere?
Or is it just passed to the `Create` function and I consequently need to handle it?